### ISSUE TITLE
Gradient checkpointing in text generation pipeline

### ISSUE THREAD

**Issue by matsuobasho** [Azure/azureml-assets#1880]
### Describe your suggestion

I'm running a job that's a version of [this ](https://github.com/Azure/azureml-assets/blob/main/assets/training/finetune_acft_hf_nlp/components/pipeline_components/text_generation/README.md)text generation pipeline.  

In the logs for my job, I see a `gradient_checkpointing=False` parameter.  However, this doesn't appear in the referenced `yaml`.  Is it a valid argument?  If so, please add to the yaml document, and if not, is there a way to add it?

**Comment by matsuobasho** [comment]
Hi @jpmann, I'm finetuning a CodeLlama model and I believe that it's not enabled in that one.  I don't have proof though because I had to delete my run results (and didn't have a screenshot of the parameters from the GUI).  Including the part of the pipeline with the inputs (excluding the path inputs since not relevant for this question):

```python
$schema: https://azuremlschemas.azureedge.net/latest/pipelineJob.schema.json
type: pipeline

experiment_name: codellama-run

settings:
  force_rerun: true
  continue_on_step_failure: false

inputs:
  mlflow_model_path:
      path: azureml://registries/azureml-meta/models/CodeLlama-7b-hf/versions/3
      text_key: "docstring"
      ground_truth_key: "function"
      batch_size: 5
      pad_to_max_seq_length: true
      max_seq_length: 500

      number_of_gpu_to_use_finetuning: 1
      num_train_epochs: 4
      per_device_train_batch_size: 2
      per_device_eval_batch_size: 2
      learning_rate: 2e-5
      gradient_accumulation_steps: 5
      eval_accumulation_steps: 5

      apply_lora: "true"
      apply_deepspeed: "true"
      apply_ort: "true"
      precision: 16
```

**Comment by matsuobasho** [comment]
@vizhur   please reopen this issue as the matter isn't resolved and I posted follow-up information.  Thanks
